lib/team_statistics.rb

```diff
@@ -118,11 +118,65 @@ def favorite_opponent(team_id)
     find_team(highest_win_percent[0].to_s).teamname
   end
 
+  # def rival(team_id)
+  #   lowest_win_percentage = win_percent_per_team(team_id).max_by do |t_id, win_percent|
+  #       win_percent
+  #   end
+  #   find_team(lowest_win_percentage[0].to_s).teamname
+  # end
+
   def rival(team_id)
-    lowest_win_percentage = win_percent_per_team(team_id).max_by do |t_id, win_percent|
-        win_percent
+    # sort the hash to find rival
+    rival_id = opponent_win_count(team_id).sort_by {|k, v| v}.last[0]
```

Comment:
I like this method, I'm curious though at the time difference between the original and the updated one?